lib/ethereumex/http_client.ex

```diff
@@ -4,7 +4,9 @@ defmodule Ethereumex.HttpClient do
   use Ethereumex.Client.BaseClient
   alias Ethereumex.Config
 
-  @spec post_request(binary(), []) :: {:ok | :error, any()}
+  @type opt :: {:url, String.t()}
```

Comment:
it also can be empty. 

I think the correct typespec for opts would be
```elixir
[] | [url: String.t()] 
```

Reply:
I'm happy to change it to your suggestion. But from my understanding, both my current approach and your suggested example are valid. My current approach also covers the case of future unordered options.

https://hexdocs.pm/elixir/typespecs.html#literals

Reply:
sounds good. just add braces. and pr's good to go

Reply:
:+1: updated with braces